src/dctl-contract-test.js

```diff
@@ -0,0 +1,181 @@
+const program = require('commander');
+const fs = require('fs');
+const shell = require('shelljs');
+const path = require('path');
+
+program
+  .description('Test a smart contract as it would be executed on an  L1 Dragonchain.', {
+    image: 'The docker image for this smart contract',
+    cmd: 'The command to run on this docker container',
+    containerArgs: '(optional) The arguments to pass in with the command for this smart contract'
+  })
+  .arguments('<image>')
+  .option('-p, --payload <payload>', 'The payload which you want to pass to your contract.')
+  .option('-d, --test-directory <testDirectory>', 'The folder which contains testing configuration files. Default: "./test"')
+  .option('-t, --tag <tag>', 'The tag you expect attached to the transaction')
+  .action((image, options) => {
+    const { payload, testDirectory } = options || {};
+
+    main()
+      .catch(e => console.error(`Error: ${e.message}`))
+      .then(killWebserver);
+
+    async function main() {
+      const { testRoot, localEnv, localHeap, localSecrets } = getPaths(testDirectory);
+      if (payload === undefined) throw new Error(`(-p) --payload is not defined.`);
+      const networkName = 'dragonchain-webserver';
+      const config = await getConfig(testRoot);
+      createDockerNetwork(networkName);
+      startWebserver(localEnv, networkName, localHeap);
+      const contractRunShell = await runContract(image, payload, networkName, config.startCommand, localEnv, localSecrets);
+      await handleContractOutput(contractRunShell.stdout, localHeap);
+    }
+  })
+  .parse(process.argv);
+
+/**
+ * handleContractOutput
+ * @param {string} stdout
+ * @param {string} localHeap
+ */
+async function handleContractOutput(stdout, localHeap) {
+  try {
+    const obj = JSON.parse(stdout);
+    if (obj.OUTPUT_TO_HEAP && obj.OUTPUT_TO_HEAP === false) {
+      console.warn('warn: OUTPUT_TO_HEAP false. Not writing to heap state.');
+    }
+    await Promise.all(Object.entries(obj).map(([key, value]) => fs.promises.writeFile(path.join(localHeap, key), value)));
+  } catch (err) {
+    const raw = path.join(localHeap, 'rawResponse');
+    console.warn(`warn: Contract output not valid JSON. Writing to "${raw}".`);
+    await fs.promises.writeFile(raw, stdout);
+  }
+}
+
+function createDockerNetwork(networkName) {
+  shell.exec(`docker network create ${networkName}`, { silent: true });
+}
+
+async function getConfig(testRoot) {
+  return JSON.parse(await fs.promises.readFile(path.join(testRoot, 'config.json'), 'utf-8'));
+}
+
+async function runContract(image, payload, network, startCommand, localEnv, localSecrets) {
+  const arrOfSecretFiles = await fs.promises.readdir('./test/secrets');
```

Comment:
should use a cross-platform way to form this path. I.e. windows uses '\', not '/'

Reply:
yupyup. fixed